pygbm/gradient_boosting.py

```diff
@@ -99,6 +100,7 @@ def fit(self, X, y):
         rng = check_random_state(self.random_state)
 
         self._validate_parameters()
+        self.n_features_ = X.shape[1]  # used for validation in predict()
```

Comment:
Instead of introducing a new attribute in the public API I would rather add a `n_features` attribute to the `TreePredictor` class to be passed as a constructor parameter and use that to do the check in the fitted `GradientBoostingMachine` classes.

Reply:
The problem is that `raw_predict` (where we check the number of features) might be called while no `TreePredictor` has been created yet: we always compute the training / validation score of the init model, before the first boosting iteration.

Reply:
Alright.